### ISSUE TITLE
Disabling lazy loading in provide

### ISSUE THREAD

**Issue by IvanKirpichnikov** [reagento/dishka#73]
__Idea:__
I suggest adding the ability to disable lazy dependency loading for the 'provide' function or globally for the entire 'Provider'

__How is it supposed to work?__
When creating a new container, create all dependencies with the 'lazy=False' flag and add them to the cache.

__How I see it in the code__
*Provide*
```py
class MyProvider(Provider)
    a = provide(A, scope=Scope.App, lazy=False)
```

*Provider*
```py
class MyProvider(Provider)
    lazy = Fasle
    a = provide(A, scope=Scope.App)
```

**Comment by Tishka17** [comment]
Note: we probably cannot implement it for async container as it requires async loop, which is often not available at creation point

**Comment by Tishka17** [comment]
We can add separate method, which creates all dependencies of current scope. As it will be called after creation it can be async

**Comment by tuukkamustonen** [comment]
Was there such a method nowadays? Something which would initialize all dependencies in a container?

Use case: Pre-load/warm a web application before taking it online.

**Comment by Tishka17** [comment]
No, there is no additional method because someone would like to specify
* which objects should be warmed
* additional actions to do 

the easiest way is to do it "manually" in your framework startup event. Just call `container.get()` with proper classes and do what you believe useful in your case (like `select 1` or other service pings). 

as an option - do it in "readyness" probe handler instead of startup

**Comment by tuukkamustonen** [comment]
> which objects should be warmed

I'd just load up it all, why skip something?

I'm sure there are use cases with more granular needs, too. But the _basic use case_ would to initialize everything, no?

In our case, we have like 50+ "main level" dependencies (these being controller classes for the endpoints), and re-declaring those would be somewhat annoying.

Of course, I could define the controllers in a separate list, which I can then feed to the provider and also read in the loop which `get()`s everything. It would be nicer to have that "load it all up" method, though.

> what you believe useful in your case (like select 1 or other service pings).

I'm actually hoping that could be more automated, too, like in https://github.com/reagento/dishka/issues/198 (I'll add a comment there, too).

> do it in "readyness" probe handler instead of startup

What would that solve?

I don't think there's benefit in `get()`'ing everything in _readiness_ probe - all of that would just come from cache.

It should be sufficient to initialize things in the startup probe, and after that the container is already warm so getting they all up again each time would not give anything (except if that triggered pings/health checks).

**Comment by tuukkamustonen** [comment]
I took a look at https://github.com/reagento/dishka/pull/114 and did something like:

```py
        for key in container.registry.factories:
            ...filtering...
            await self.adi._get(key)  # noqa: SLF001
```

Seems functional and quite simple.
